soc/arm/ti_k3/j721e/soc.c

```diff
@@ -0,0 +1,97 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/fatal.h>
+
+#include "ipc_rpmsg_linux_resource_table.h"
+
+#define DEBUG_MEM_LOG_SIZE 1024
+
+__attribute__((section(".log_shared_mem"))) char volatile gdebugmemlog[DEBUG_MEM_LOG_SIZE];
+
+const struct rpmessage_resourcetable grpmessage_linuxresourcetable __aligned(4096)
+	__attribute__ ((section(".resource_table"))) = {
+	{
+		1U,         /* we're the first version that implements this */
+		2U,         /* number of entries, MUST be 2 */
+		{ 0U, 0U, } /* reserved, must be zero */
+	},
+	/* offsets to the entries */
+	{
+		offsetof(struct rpmessage_resourcetable, vdev),
+		offsetof(struct rpmessage_resourcetable, trace),
+	},
+	/* vdev entry */
+	{
+		RPMESSAGE_RSC_TYPE_VDEV, RPMESSAGE_RSC_VIRTIO_ID_RPMSG,
+		0U, 1U, 0U, 0U, 0U, 2U, { 0U, 0U },
+	},
+	/* the two vrings */
+	{ RPMESSAGE_RSC_VRING_ADDR_ANY, 4096U, 256U, 1U, 0U },
+	{ RPMESSAGE_RSC_VRING_ADDR_ANY, 4096U, 256U, 2U, 0U },
+	{
+		(RPMESSAGE_RSC_TRACE_INTS_VER0 | RPMESSAGE_RSC_TYPE_TRACE),
+		(uint32_t)gdebugmemlog, DEBUG_MEM_LOG_SIZE,
+		0, "trace:r5fss0_0",
+	},
+};
+
+static int j721e_init(void)
+{
+	return 0;
+}
+
+uint32_t sys_clock_elapsed(void)
+{
+	return 0;
+}
+
+uint32_t sys_clock_cycle_get_32(void)
+{
+	return 0;
+}
+
+uint64_t sys_clock_cycle_get_64(void)
+{
+	return 0;
+}
+
+void z_soc_irq_init(void)
+{
+}
+
+void z_soc_irq_enable(unsigned int irq)
+{
+}
+
+void z_soc_irq_disable(unsigned int irq)
+{
+}
+
+int z_soc_irq_is_enabled(unsigned int irq)
+{
+	return 0;
+}
+
+void z_soc_irq_priority_set(unsigned int irq, unsigned int prio, unsigned int flags)
+{
+}
```

Comment:
Sorry, this is not acceptable. A full interrupt controller driver is definitely needed before the SoC is considered for inclusion in-tree.

Reply:
I will add the interrupt controller support as part of this PR.

Reply:
@slpp95prashanth can you also please mark the PR as draft until then?